### ISSUE TITLE
[Sequencer] Ansible test failure on txpool race condition test

### ISSUE THREAD

**Issue by mandrigin** [0xPolygon/cdk-erigon#621]
```
ansible_1  | TASK [pool-race-conditions : Recollecting test ETH] ****************************
ansible_1  | fatal: [localhost]: FAILED! => changed=true 
ansible_1  |   cmd: |-
ansible_1  |     pushd /tmp/rpc-tests.d || exit 1
ansible_1  |     sleep 2
ansible_1  |     addr=$(jq -r '.[0].address' /tmp/rpc-tests.d/pool-test-wallet.json)
ansible_1  |     tot_wei=$(cast balance --block pending --rpc-url *** $addr)
ansible_1  |     gas_price=$(cast gas-price --rpc-url ***)
ansible_1  |     cast send --legacy --value $((tot_wei - gas_price * 21000)) --rpc-url *** --private-key $(jq -r '.[0].private_key' /tmp/rpc-tests.d/pool-test-wallet.json) 0x41BB6960a5156aC29d6e9E04273837AD19d6691A
ansible_1  |   delta: '0:00:03.359657'
ansible_1  |   end: '2024-06-13 16:20:37.130455'
ansible_1  |   msg: non-zero return code
ansible_1  |   rc: 1
ansible_1  |   start: '2024-06-13 16:20:33.770798'
ansible_1  |   stderr: |-
ansible_1  |     Error:
ansible_1  |     server returned an error response: error code -32000: RPC error response: INTERNAL_ERROR: could not replace existing tx
ansible_1  |   stderr_lines: <omitted>
ansible_1  |   stdout: /tmp/rpc-tests.d /ansible
ansible_1  |   stdout_lines: <omitted>
ansible_1  | 
```

**Comment by mandrigin** [comment]
From John

```
I think ...

- Take an account that has balance B and current nonce n
- Send a valid transaction for the next nonce (n+1) that uses more than half of the balance B/2 + 1
- Send a failing transactions for the current nonce (n) that also uses more than half of the balance B/2 + 1

Expected results:
It's impossible for both of these transactions to succeed. That's the point. It creates a scenario where you'll have two transactions in the pool that if one suceedes, the other must fail. After sending both transactions, both transactions are mined. The n transaction is mined as reverting and therefore the B/2 + 1 is not used. After n is mined, n+1 can mined and it will work properly because our total balance B is still available.

What's actually happening:
I'm not fully sure. The details are in the screenshot. The n transaction is mined appropriately, but it seems like n+1 is perpetually stuck in queued and ... never gets mined. Even if I resend the transaction, it doesn't get mined (ever?). The easiest solution to bypass this scenario is to send ANOTHER transaction n+2  which forces both n+1 and n+2 into pending and it's mined.
```

**Comment by mandrigin** [comment]
![IMG_0249](https://github.com/0xPolygonHermez/cdk-erigon/assets/466427/51d1bcff-287e-4935-b818-16371b57cae8)

**Comment by kstoykov** [comment]
I started by checking the very first step using latest commit of zkevm branch. I had no problem getting the transaction using "cast tx". Could you have a look on the screen shot and let me know what have I missed.

![image](https://github.com/0xPolygonHermez/cdk-erigon/assets/24619432/47e4e112-c4c1-477b-8ec8-0a559911a239)

**Comment by mandrigin** [comment]
should be fixed in #303

**Comment by mandrigin** [comment]
Fix confirmed, the test now passes
